Mathlib/RingTheory/MvPolynomial/WeightedHomogeneous.lean

```diff
@@ -455,14 +489,66 @@ theorem weightedHomogeneousComponent_zero [NoZeroSMulDivisors ℕ M] (hw : ∀ i
   rcases Classical.em (d = 0) with (rfl | hd)
   · simp only [coeff_weightedHomogeneousComponent, if_pos, map_zero, coeff_zero_C]
   · rw [coeff_weightedHomogeneousComponent, if_neg, coeff_C, if_neg (Ne.symm hd)]
-    simp only [weightedDegree', LinearMap.toAddMonoidHom_coe, Finsupp.total_apply, Finsupp.sum,
+    simp only [weightedDegree, LinearMap.toAddMonoidHom_coe, Finsupp.total_apply, Finsupp.sum,
       sum_eq_zero_iff, Finsupp.mem_support_iff, Ne.def, smul_eq_zero, not_forall, not_or,
       and_self_left, exists_prop]
     simp only [FunLike.ext_iff, Finsupp.coe_zero, Pi.zero_apply, not_forall] at hd
     obtain ⟨i, hi⟩ := hd
     exact ⟨i, hi, hw i⟩
 #align mv_polynomial.weighted_homogeneous_component_zero MvPolynomial.weightedHomogeneousComponent_zero
 
+/-- A weight function is nontrivial if its values are not torsion. -/
+def NonTrivialWeight (w : σ → M) :=
+  ∀ n x, n • w x = (0 : M) → n = 0
+
+theorem nonTrivialWeight_of [NoZeroSMulDivisors ℕ M] (hw : ∀ i : σ, w i ≠ 0) :
+    NonTrivialWeight w := by
+  intro n x
+  rw [smul_eq_zero]
+  intro hnx
+  cases' hnx with hn hx
+  · exact hn
+  · exact absurd hx (hw x)
+#align mv_polynomial.non_trivial_weight_of MvPolynomial.nonTrivialWeight_of
```

Comment:
I can't find the string `non_trivial_weight_of` in mathlib 3, should this `#align` even exist?
```suggestion
```

Reply:
You are right, this never made it to mathlib 3. I am also renaming `NonTrivialWeight` to `NonTorsionWeight`, following @AntoineChambert-Loir's suggestion.